cli: TestZipIncludeAndExcludeFilesDataDriven failed

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21273786?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21273786?buildTab=artifacts#/) on master @ [02e602231d073faf1deb1c8b3d26f1422ba873ca](https://github.com/cockroachdb/cockroach/commits/02e602231d073faf1deb1c8b3d26f1422ba873ca):


```
debug/system.rangelog.txt
debug/system.region_liveness.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_hints.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_metadata.txt
debug/system.table_statistics.txt
debug/system.table_statistics_locks.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/system.transaction_diagnostics.txt
debug/system.transaction_diagnostics_requests.txt
debug/system.zones.txt
debug/tenant_ranges.err.txt
    zip_test.go:1602: -- test log scope end --
    zip_test.go:1540: 
          testdata/zip/file-filters/testzip_file_filters:2:
          --include-files=* --exclude-files=cpu.pprof
        output didn't match expected:
          @@ -69,10 +69,12 @@
           debug/nodes/1/crdb_internal.node_txn_execution_insights.txt
           debug/nodes/1/crdb_internal.node_txn_stats.txt
           debug/nodes/1/details.json
           debug/nodes/1/gossip.json
           debug/nodes/1/heap.pprof
          +debug/nodes/1/heapprof/ash_report.2026-03-20T00_13_57.764.goroutine_dump.json
          +debug/nodes/1/heapprof/ash_report.2026-03-20T00_13_57.764.goroutine_dump.txt
           deterministic.log
           debug/nodes/1/lsm.txt
           debug/nodes/1/ranges.json
           debug/nodes/1/stacks.pprof
           debug/nodes/1/stacks.txt
--- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (23.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164968 cli: TestZipIncludeAndExcludeFilesDataDriven failed [C-test-failure O-robot T-observability branch-release-26.1 s390x-test-failure]
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61812

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -1521,6 +1521,9 @@ func trimNonDeterministicZipOutputFiles(out string) string {
 	out = re.ReplaceAllString(out, ``)
 	re = regexp.MustCompile(`(?m).*goroutine_dump.*\.pb\.gz$` + "\n")
 	out = re.ReplaceAllString(out, ``)
+	// ASH report files have timestamps in filenames, making them non-deterministic.
+	re = regexp.MustCompile(`(?m).*ash_report.*\.(json|txt)$` + "\n")
+	out = re.ReplaceAllString(out, ``)
 	// CPU profiles are non-deterministic: stored profiles have timestamps in
 	// filenames, and live profile collection may fail producing error files.
 	re = regexp.MustCompile(`(?m).*cpuprof/cpuprof\..*\.pprof$` + "\n")

